### ISSUE TITLE
Improve information for customizing dependency versions

### ISSUE THREAD

**Issue by terribleherbst** [spring-projects/spring-boot#10141]
The description of customizing dependency should be improved:

Currently changing dependency version is described here:
https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#howto-customize-dependency-versions 

The information is pretty short, it would be great do have some more detailed information. For gradle it would be great to know where the user should place the `ext['slf4j.version'] = '1.7.5'` part.
The best practice addressed in this stackoverflow question (Suggested Steps at ) https://stackoverflow.com/questions/45967464/how-to-selectively-upgrade-a-dependency-in-spring-boot-sample-case-spring-data might be more helpful than _Overriding versions may cause compatibility issues_.

**Comment by wilkinsona** [comment]
As I said on Twitter, I don't think a discussion about best practices for upgrading dependencies belongs in Spring Boot's documentation. Spring Boot's documentation is already quite long so we need to be careful about what is and is not out of scope to avoid it becoming unmanageable. I think that general advice (that isn't Boot or even Java-specific) about how to upgrade a dependency is out of scope.

I do, however, consider describes the mechanics of upgrading a dependency to be in scope, but I think we've already covered that quite well. There's no mention of where to specifically place `ext['slf4j.version'] = '1.7.5'` as it can go anywhere in `build.gradle`. The documentation also links to the dependency management plugin's documentation which [describes this in more detail](https://github.com/spring-gradle-plugins/dependency-management-plugin#changing-the-value-of-a-version-property).

With all that said, if you have a suggestion for something that would address your concern in a concise manner that considers the points made above then we can certainly consider it.

**Comment by spring-issuemaster** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by terribleherbst** [comment]
Ok I agree best or good practice might not belong into the core spring boot documentation, but I guess such information is valueable. Maybe there is some other place?

Regarding the information about gradle. Just mentioning that "the information could be place anywhere in build.gradle" is just one more sentence and can help beginners a lot.

**Comment by wilkinsona** [comment]
We try to be consistent, so if we add the sentence that you have proposed above, we'd also have to add similar sentences for other technologies in tens of others places. The content would get unwieldy if we did that. In short, for the reference documentation at least, we have to assume a certain base level of knowledge.

I think a guide on https://spring.io/guides may be a better place for the content that you are proposing. I've opened [an issue](https://github.com/spring-guides/getting-started-guides/issues/52) to discuss the possibility.
